src/Mapbender/CoreBundle/Resources/public/mapbender.element.searchRouter.js

```diff
@@ -398,37 +401,66 @@
         /**
          * Rebuilds result table with search result data.
          *
-         * @param SearchModel       model   Search model
-         * @param FeatureCollection results Search result feature collection
-         * @param object            options Backbone options
+         * @param {SearchModel} model
+         * @param {FeatureCollection} results
+         * @param {Object} options Backbone options (not used?)
          */
         _searchResultsTable: function(model, results, options){
             var headers = this.options.routes[this.selected].results.headers,
                 table = $('.search-results table', this.element),
                 tbody = $('<tbody></tbody>'),
-                layer = this._getLayer(true);
+                layer = this._getLayer(true),
+                self = this;
 
             $('tbody', table).remove();
             layer.removeAllFeatures();
-            features = [];
+            var features = [];
 
             if(results.length > 0) $('.no-results', this.element).hide();
 
-            results.each(function(feature, idx){
+            results.each(function(feature, idx) {
                 var row = $('<tr/>');
                 row.addClass(idx % 2 ? "even" : "odd");
                 row.data('feature', feature);
-                for(var header in headers){
+
+                for (var header in headers) {
                     var d = feature.get('properties')[header];
                     row.append($('<td>' + (d || '') + '</td>'));
                 }
+
                 tbody.append(row);
 
                 features.push(feature.getFeature());
             });
 
             table.append(tbody);
             layer.addFeatures(features);
+
+            $('.search-results tbody tr')
+                .on('click', function () {
+                    var feature = $(this).data('feature').getFeature();
+                    self._highlightFeature(feature, layer, 'select');
+                })
+                .on('mouseenter', function () {
+                    var feature = $(this).data('feature').getFeature();
+
+                    if(feature.renderIntent !== 'select') {
+                        self._highlightFeature(feature, layer, 'temporary');
+                    }
+                })
+                .on('mouseleave', function () {
+                    var feature = $(this).data('feature').getFeature();
+
+                    if(feature.renderIntent !== 'select') {
+                        self._highlightFeature(feature, layer, 'default');
+                    }
+                })
+            ;
+        },
+
+        _highlightFeature: function (feature, layer, style) {
+            feature.renderIntent = style;
+            layer.redraw();
```

Comment:
Why not simply use `feature.layer.redrawFeature(feature, style)`?
The `layer` argument is redundant.
Actually, the entire method is redundant.

Reply:
Thats something I wasnt beware of. I will take a look on `redrawFeature` and give you feedback again.

Reply:
After researching about `drawFeature`, I came to the conclusion that it should be used instead of settings the renderIndent. [OL2 drawFeature() Doc](http://dev.openlayers.org/apidocs/files/OpenLayers/Layer/Vector-js.html#OpenLayers.Layer.Vector.drawFeature)

Reply:
Sorry, my mistake, it's [drawFeature](https://github.com/openlayers/ol2/blob/75716beaacc01955f05708c03438b588fc8db55d/lib/OpenLayers/Layer/Vector.js#L788), not redrawFeature.